### ISSUE TITLE
Do not allow placement of Query Loop inside Post Content

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#40944]
Query Loop as it exists today has kind of been built as a block-based interpretation of `WP_Query`. While powerful, it has inevitably resulted in an overwhelming UX. For example the confusing and unpredictable "Inherit query from template" option exists because it's possible to place Query Loop in templates _and_ posts. Not to mention that blocks like Query Pagination simply do not work in posts at all, with no explanation offered by the interface.

<img width="280" alt="Screenshot 2022-05-09 at 16 52 01" src="https://user-images.githubusercontent.com/846565/167448523-921ac148-9570-4196-9175-5838b4861d3f.png">

Before this gets too far out of hand, it might be good to explore contextually optimised implementations, based on the various jobs users employ Query blocks to perform.

By prohibiting the placement of the base Query block in Post Content it immediately becomes singular in purpose – display posts as defined by the template. Consequently it needn't entertain so many options and the overall experience can be streamlined.

<img width="280" alt="Screenshot 2022-05-09 at 16 55 44" src="https://user-images.githubusercontent.com/846565/167449237-eca23132-5714-4cd3-87c4-05b505d70696.png">

Naturally we still need to provide ways to display post lists in pages and so on, but these use cases can be handled by simple variations of the base Query Loop as discussed in:

* https://github.com/WordPress/gutenberg/issues/32268
* https://github.com/WordPress/gutenberg/issues/40941

It goes without saying that we cannot implement this change until those variations exist. We'd also need to consider how to handle pre-existing Query Loops as well.

**Comment by mtias** [comment]
I don't think we should exclude query block from post-content until we have the variations in place.

Edit: I see you mention the same thing at the end :)
